.github/workflows/release.yml

```diff
@@ -181,3 +181,91 @@ jobs:
         name: macos-unsigned-pkg
         path: |
           pkg
+  
+  sign-and-notarize-macos:
+    name: Sign and notarize macOS package file
+    # ESRP service requires signing to run on Windows
+    runs-on: windows-latest
+    needs: pack
+    steps:
+    - name: Check out repository
+      uses: actions/checkout@v3
+
+    - name: Download unsigned package file
+      uses: actions/download-artifact@v2
+      with:
+        name: macos-unsigned-pkg
+        path: pkg
+    
+    - name: Zip unsigned package file
+      shell: pwsh
+      run: |
+        Compress-Archive -Path pkg/*.pkg pkg/gcm-pkg.zip
+        cd pkg
+        Get-ChildItem -Exclude gcm-pkg.zip | Remove-Item -Recurse -Force
+    
+    - uses: azure/login@v1
+      with:
+        creds: ${{ secrets.AZURE_CREDENTIALS }}
+    
+    - name: Install ESRP client
+      shell: pwsh
+      env:
+        AZ_KEY: ${{ secrets.AZURE_STORAGE_KEY }}
+      run: |
+        az storage blob download --file esrp.zip --account-key "$env:AZ_KEY" --account-name gcmesrp --container microsoft-esrp-client --name microsoft.esrpclient.1.2.76.nupkg
+        Expand-Archive -Path esrp.zip -DestinationPath .\esrp
+    
+    - name: Install certificates
+      shell: pwsh
+      env:
+        AZ_VAULT: ${{ secrets.AZURE_VAULT }}
+        AUTH_CERT: ${{ secrets.AZURE_VAULT_AUTH_CERT_NAME }}
+        REQUEST_SIGNING_CERT: ${{ secrets.AZURE_VAULT_REQUEST_SIGNING_CERT_NAME }}
+      run: |
+        az keyvault secret download --vault-name "$env:AZ_VAULT" --name "$env:AUTH_CERT" --file out.pfx
+        certutil -f -importpfx out.pfx
+        Remove-Item out.pfx
+
+        az keyvault secret download --vault-name "$env:AZ_VAULT" --name "$env:REQUEST_SIGNING_CERT" --file out.pfx
+        certutil -f -importpfx out.pfx
+        Remove-Item out.pfx
```

Comment:
Regarding reusable parts of the workflow.. if GitHub Workflows aren't the best at reusability, perhaps consider moving this part to a shell (Power)script in the repository that each signing job can just run with the appropriate args/envars?

```yaml
    - name: Set up ESRP client
      shell: pwsh
      run: |
        .github\setup-esrp.ps1 `
            -KeyVaultName ${{ secrets.AZURE_VAULT }} `
            -AuthCertName ${{ secrets.AZURE_VAULT_AUTH_CERT_NAME }} `
            -SignCertName ${{ secrets.AZURE_VAULT_REQUEST_SIGNING_CERT_NAME }}
```

Reply:
Good idea! Will work on writing that up today.